core/src/main/java/org/apache/calcite/sql/type/SqlTypeConstructorOperandTypeChecker.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.type;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlOperator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Parameter type-checking strategy for SqlTypeConstructor
+ */
+public class SqlTypeConstructorOperandTypeChecker implements SqlOperandTypeChecker {
+  //~ Methods ----------------------------------------------------------------
+
+  private RelDataType type;
+
+  public SqlTypeConstructorOperandTypeChecker(RelDataType type) {
+    this.type = type;
+  }
+
+  public boolean isOptional(int i) {
+    return false;
+  }
+
+  public boolean checkOperandTypes(
+      SqlCallBinding callBinding,
+      boolean throwOnFailure) {
+    List<SqlTypeFamily> families = new ArrayList<>();
+
+    List<RelDataTypeField> fieldList = type.getFieldList();
+    for (int i = 0; i < fieldList.size(); i++) {
+      RelDataTypeField field = fieldList.get(i);
+      SqlTypeName sqlTypeName = field.getType().getSqlTypeName();
+      if (sqlTypeName == SqlTypeName.ROW) {
+        if (field.getType().equals(callBinding.getOperandType(i))) {
+          families.add(SqlTypeFamily.ANY);
```

Comment:
Why the nested ROW type is coerced to `ANY` ?

Reply:
Please correct me if I'm wrong SqlOperandTypeChecker supports SqlTypeFamily. Is there a way to validate RelDataType? Couldnot find any reference for validating RelRecordType during function parameter checking

Reply:
Just use `SqlTypeName.ROW` you can define a customized family checker by yourself.

Reply:
I think we should also check operand type for `SqlTypeName.ROW`.

Reply:
@ritesh-kapoor Thanks, we can support the nested `ROW` validation in the near future.